components/regionDropdowns/LocaleDropdown.tsx

```diff
@@ -0,0 +1,56 @@
+import { useRouter } from "next/router";
+import React from "react";
+
+import { LOCALES } from "@/lib/regions";
+
+import { useRegions } from "../RegionsProvider";
+import { BaseRegionsDropdown } from "./BaseRegionsDropdown";
+import { BaseRegionsDropdownItem } from "./BaseRegionsDropdownItem";
+
+interface DropdownOption {
+  label: string;
+  chosen: boolean;
+  localeSlug: string;
+}
+
+export function LocaleDropdown() {
+  const router = useRouter();
+  const { currentLocale, currentChannel } = useRegions();
+
+  const localeOptions: DropdownOption[] = LOCALES.map((loc) => ({
+    label: loc.name,
+    chosen: loc.slug === currentLocale,
+    localeSlug: loc.slug,
+  }));
+
+  const onLocaleChange = async (localeSlug: string) => {
```

Comment:
```suggestion
  const onLocaleChange = (localeSlug: string) => {
```
remove awaits inside